### PR TITLE
Add full TS support to front end

### PR DESCRIPTION
**Pull request by affordances**

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/components/component_kit/cw_content_page.tsx [hunk 1]
```diff
@@ -36,7 +36,7 @@ type ContentPageAttrs = {
   author?: m.Vnode;
   actions?: Array<MenuItem>;
   body?: m.Vnode;
-  comments?: Array<Comment<any>>;
+  comments?: m.Vnode;
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/components/component_kit/cw_content_page.tsx:39]: Reason for this change?

File: packages/commonwealth/client/scripts/views/components/proposal_card/helpers.tsx [hunk 2]
```diff
@@ -184,13 +184,10 @@ export const getStatusText = (proposal: AnyProposal) => {
   return '';
 };
 
-export const getPrimaryTagText = (proposal: AnyProposal) => [
-  chainEntityTypeToProposalShortName(
+export const getPrimaryTagText = (proposal: AnyProposal) => `
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/components/proposal_card/helpers.tsx:187]: Does this array -> string change affect anything?

File: packages/commonwealth/client/scripts/views/components/sidebar/governance_section.tsx [hunk 3]
```diff
@@ -479,7 +475,7 @@ export class GovernanceSection
       },
       displayData: governanceGroupData,
       isActive: false,
-      toggleDisabled: vnode.attrs.mobile,
+      toggleDisabled: false,
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/components/sidebar/governance_section.tsx:478]: Does this affect anything?

File: packages/commonwealth/client/scripts/views/pages/view_thread/index.tsx [hunk 4]
```diff
@@ -629,73 +632,75 @@ class ViewThreadPage
...
-                ]
-              : []),
-          ]}
+                        </div>
+                      ),
+                    },
+                  ]
+                : []),
+              ...(this.polls?.length > 0 ||
+              (isAuthor && (!app.chain?.meta?.adminOnlyPolling || isAdmin))
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/pages/view_thread/index.tsx:677]: These changes safe? Seem ok just want to make sure.

**Review by jnaviask — COMMENTED** [review]
Really great work. Flagged a couple spots for @zakhap to confirm, but I feel very comfortable getting this out.

File: packages/commonwealth/client/scripts/views/components/component_kit/cw_content_page.tsx (hunk 1 above)
Comment (affordances) [reply packages/commonwealth/client/scripts/views/components/component_kit/cw_content_page.tsx:39]: It was left over from an earlier factoring.

File: packages/commonwealth/client/scripts/views/components/proposal_card/helpers.tsx (hunk 2 above)
Comment (affordances) [reply packages/commonwealth/client/scripts/views/components/proposal_card/helpers.tsx:187]: It's supposed to return a string, so the only change is to make it pass the type check.

Comment (affordances) [reply packages/commonwealth/client/scripts/views/components/proposal_card/helpers.tsx:187]: (There's no reason for it to return an array as opposed to a string interpolation)

File: packages/commonwealth/client/scripts/views/components/sidebar/governance_section.tsx (hunk 3 above)
Comment (affordances) [reply packages/commonwealth/client/scripts/views/components/sidebar/governance_section.tsx:478]: `vnode.attrs.mobile` doesn't exist and hasn't for a while, so `false` is equivalent 🤷 .

File: packages/commonwealth/client/scripts/views/pages/view_thread/index.tsx (hunk 4 above)
Comment (affordances) [reply packages/commonwealth/client/scripts/views/pages/view_thread/index.tsx:677]: Yeah, IIRC that was just to cast the array to `SidebarComponents`.

**Review by jnaviask — APPROVED** [review]

**Pull request merged**